community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java

```diff
@@ -573,4 +510,100 @@ private void assertOfOneOfTypes( PropertyType type1, PropertyType type2 )
             throw new IllegalStateException( "Expected type " + type1 + " or " + type2 + " but was " + this.type );
         }
     }
+
+    private static class Block
+    {
+        public static final int VALUES_SIZE = DEFAULT_PAYLOAD_SIZE / 8;
+
+        private int writeIndex;
+        private long[] values = new long[VALUES_SIZE];
+        private int remaining;
+
+        public void init()
+        {
+            writeIndex = 0;
+            remaining = 0;
```

Comment:
I can do that, but what is the benefit? I think I made sure we read only something that has been written, no?